angular-dependancy-injection/dependancy-injection/src/app/expense-table/expense-table.component.ts

```diff
@@ -0,0 +1,127 @@
+import { Component, OnInit, } from '@angular/core';
+import { ExpensesService } from '../core/services/expenses.service';
+
+
+@Component({
+  selector: 'expense-table',
+  templateUrl: './expense-table.component.html',
+  styleUrls: ['./expense-table.component.scss']
+})
+export class ExpenseTableComponent implements OnInit {
+
+  constructor(private exp: ExpensesService) { }
+
+  exepense_month_january: number = 0;
+  exepense_month_february: number = 0;
+  exepense_month_march: number = 0;
+  exepense_month_april: number = 0;
+  exepense_month_may: number = 0;
+  exepense_month_june: number = 0;
+  // totalExpense =
+  //   this.exepense_month_january +
+  //   this.exepense_month_february +
+  //   this.exepense_month_march +
+  //   this.exepense_month_april +
+  //   this.exepense_month_may +
+  //   this.exepense_month_june;
+  totalExpense = 0;
+  current_year: number = 2022;
+  //array_of_month = [0, 1, 2, 3, 4, 5];
+  //min_dates_for_entry: any = 2022 - 01 - 01;
+  onlySign = "/-";
+  web_hint = "Add an expense to get started.";
+  getDataArray: any =
+    {
+      expense_title: '',
+      expense_date: '',
+      expense_amount: ''
+    }
+
+  expenses: { expense_title: string; expense_date: Date; expense_amount: number }[] = [];
+
+  addExpensesApp() {
+    if (this.getDataArray.expense_title == '') {
+      alert("Expense Name Cannot Be Empty!");
+      return
+    }
+    else if (this.getDataArray.expense_date == '') {
+      alert("Expense Date Cannot Be Empty!");
+      return
+    }
+    else if (this.getDataArray.expense_amount == '') {
+      alert("Expense Amount Cannot Be Empty!");
+      return;
+    }
+    else {
+      let add: any =
+      {
+        expense_title: this.getDataArray.expense_title,
+        expense_date: this.getDataArray.expense_date,
+        expense_amount: this.getDataArray.expense_amount
+      }
+      this.exp.addExpenses(add);
+    }
+  }
+
+  resetForm() {
+    this.getDataArray = [];
+  }
+  deleteEntry(id: number) {
```

Comment:
Naming could have been better, I would have call this as deleteExpense as this the functionality of the method

Reply:
Done, will push with other changes at once.